src/shared/api/product/api.ts

```diff
@@ -0,0 +1,21 @@
+import { instance, API_ENDPOINTS } from '@/shared/api/config';
+import { queryStringify } from '@/shared/api/lib';
+
+import type {
+  TMachineriesRequestData,
+  TMachineriesResponseData,
+  TMachineryResponseData,
+  TMakeFavoriteResponseData,
+} from './typing';
+
+export const productAPI = {
+  getMachineries: (data: TMachineriesRequestData) => {
+    const url = queryStringify(data);
+    return instance.get<TMachineriesResponseData>(`${API_ENDPOINTS.machinery}/${url}/`);
+  },
+  getMachinery: (id: number) =>
+    instance.get<TMachineryResponseData>(`${API_ENDPOINTS.machinery}/${id}/`),
```

Comment:
```suggestion
  getMachinery: async (id: number) =>
    await instance.get<TMachineryResponseData>(`${API_ENDPOINTS.machinery}/${id}/`),
```
Предлагаю все get-функции сделать асинхронными. Иначе instance вернёт промис а не ответ. Промис, конечно можно обработать и потом... но так обычно не делается.